pkg/operator/operator.go

```diff
@@ -99,25 +101,45 @@ func New(config operatorconfig.Config, dnsManager dns.Manager, kubeConfig *rest.
 	if err != nil {
 		return nil, fmt.Errorf("failed to get API Group-Resources")
 	}
-	ingressCache, err := cache.New(kubeConfig, cache.Options{Namespace: "openshift-ingress", Scheme: scheme, Mapper: mapper})
+	operandCache, err := cache.New(kubeConfig, cache.Options{Namespace: "openshift-ingress", Scheme: scheme, Mapper: mapper})
 	if err != nil {
 		return nil, fmt.Errorf("failed to create openshift-ingress cache: %v", err)
 	}
-
-	for _, obj := range []runtime.Object{
+	// Any types added to the list here will only queue a clusteringress if the
+	// resource has the expected label associating the resource with a
+	// clusteringress.
+	for _, o := range []runtime.Object{
 		&appsv1.Deployment{},
 		&corev1.Service{},
 	} {
-		informer, err := ingressCache.GetInformer(obj)
+		// TODO: may not be necessary to copy, but erring on the side of caution for
+		// now given we're in a loop.
+		obj := o.DeepCopyObject()
```

Comment:
Since you don't reference `o` inside the `ToRequests` function or any other function that could capture it, I don't see any way using `o` directly could be problematic.

Reply:
If you dig deep enough through `GetInformer()` (which does take `obj`) you'll see that `obj` becomes state of at least one struct (`SharedIndexInformer`) which I'm afraid will ultimately be used in some concurrent context... anyway, I haven't done a thorough audit and there's a _ton_ of code `obj` passes through one way or another, so why not just copy it and not have to think about it? There are no performance considerations here...